models/eval/inspirational_generation.py

```diff
@@ -260,26 +266,27 @@ def resetVar(newVal):
         if lambdaD > 0:
 
             loss = -lambdaD * model.netD(noiseOut)[:, 0]
-            sumLoss += loss
+            combinedLoss[2] += loss
 
             if not randomSearch:
                 loss.sum(dim=0).backward()
 
         if nevergrad:
             for i in range(nImages):
-                optimizers[i].tell(inps[i], float(sumLoss[i]))
+                thelosses = (float(combinedLoss[0][i]), float(combinedLoss[1][i]), float(combinedLoss[2][i]))
+                optimizers[i].tell(target.compute_aggregate_loss(thelosses, inps[0]))
         elif not randomSearch:
             optimNoise.step()
 
         if optimalLoss is None:
             optimalVector = deepcopy(varNoise)
-            optimalLoss = sumLoss
+            optimalLoss = combinedLoss
 
         else:
-            optimalVector = torch.where(sumLoss.view(-1, 1) < optimalLoss.view(-1, 1),
+            optimalVector = torch.where(combinedLoss.view(-1, 1) < optimalLoss.view(-1, 1),
```

Comment:
you probably want combinedLoss.sum(dim=0) and same thing for optimalLoss here.
Otherwise you're taking parts of varNoise and parts of optimalLoss but those parts are not related to the losses that are better.